pkg/models/context.go

```diff
@@ -80,6 +85,27 @@ func (ctx *ReqContext) JsonApiErr(status int, message string, err error) {
 	ctx.JSON(status, resp)
 }
 
+// WriteErr writes an error response based on errutil.Error.
+// If provided error is not a errutil.Error a 500 response is written.
+func (ctx *ReqContext) WriteErr(err error) {
+	grafanaErr := &errutil.Error{}
+	if !errors.As(err, grafanaErr) {
+		ctx.JsonApiErr(http.StatusInternalServerError, "", fmt.Errorf("unexpected error type [%s]: %w", reflect.TypeOf(err), err))
+	}
+	ctx.JsonApiErr(grafanaErr.Reason.Status().HTTPStatus(), grafanaErr.Public().Message, err)
```

Comment:
The entire `Public` structure should be provided to the end user, not just `Message` as that will omit all details about the error other than the default message, you'll want to use `ctx.JSON` directly (and, less importantly, use the errutil log level to log the error message).